src/ray/core_worker/reference_count.cc

```diff
@@ -1003,6 +1009,43 @@ absl::optional<LocalityData> ReferenceCounter::GetLocalityData(
   return locality_data;
 }
 
+void ReferenceCounter::PushToLocationSubscribers(ReferenceTable::iterator it,
+                                                 absl::ReleasableMutexLock &lock) {
+  auto callbacks = it->second.location_subscription_callbacks;
+  it->second.location_subscription_callbacks.clear();
+  auto locations = it->second.locations;
+  auto object_size = it->second.object_size;
+  auto current_version = ++it->second.location_version;
+  // NOTE: We release the lock before running the callbacks.
+  lock.Release();
```

Comment:
Do we really need to release the lock here given that each of these callbacks will be posted to the IO event loop?

Reply:
Can you make this change?